modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java

```diff
@@ -200,50 +202,66 @@ protected void addAssociationForMatch() throws Exception {
 
     protected void waitForWorkflowToDeploy() throws Exception {
 
-        // We have to check whether the service is up and running by calling the generated endpoint.
-        String url = super.getBackendURL() + addUserWorkflowName + "Service";
-        HttpClient client = new HttpClientFactory().getHttpClient();
-        HttpGet request = new HttpGet(url);
-
-        boolean runLoop;
-        int count = 1;
-
-        do {
-            runLoop = false;
-            HttpResponse httpResponse = client.execute(request);
-
-            // If the server response is 500 or it contains "service not available" text or "Operation not found" text,
-            // then we have to assume that the service is still not available. So we have to recheck after
-            // brief time period.
-            if (httpResponse.getStatusLine().getStatusCode() == 500) {
-                runLoop = true;
-            } else {
-                BufferedReader bufferedReader =
-                        new BufferedReader(new InputStreamReader(httpResponse.getEntity().getContent()));
-                String line;
-                while ((line = bufferedReader.readLine()) != null) {
-                    if (line.contains("The service cannot be found for the endpoint reference") ||
-                            line.contains("The endpoint reference (EPR) for the Operation not found")) {
-                        runLoop = true;
-                        break;
-                    }
+        boolean isBpelDeployed = false;
+        boolean isHumanTaskDeployed = false;
+        String url;
+        for (int count = 1; count <= MAX_WAIT_ITERATIONS_TILL_WORKFLOW_DEPLOYMENT; count++) {
+            log.info("Verifying workflow deployment on " + count + " of " +
+                    MAX_WAIT_ITERATIONS_TILL_WORKFLOW_DEPLOYMENT + " attempts.");
+            if (!isBpelDeployed) {
+                log.info("Verifying BPEL deployment.");
+                url = identityContextUrls.getSecureServiceUrl() + "/" + addUserWorkflowName + "Service?wsdl";
+                if (isWorkflowDeployed(url)) {
+                    isBpelDeployed = true;
+                    log.info("BPEL Workflow deployed successfully in " + count + " attempt(s).");
```

Comment:
better reword the log. we are not deploying it here. we are just checking for the deployment completion.

Reply:
addressed by af0f52a25c490ea7670ae5b3a8b7628cc45b903a